src/crypto/vrf.rs

```diff
@@ -0,0 +1,190 @@
+use std::fmt::{Display, Formatter};
+
+use schnorrkel::keys::{MINI_SECRET_KEY_LENGTH, SECRET_KEY_LENGTH};
+use schnorrkel::vrf::{VRFInOut, VRFOutput, VRFProof};
+use schnorrkel::{ExpansionMode, MiniSecretKey, SecretKey, SignatureResult};
+
+pub struct VRFPublicKey(pub schnorrkel::keys::PublicKey);
+
+impl AsRef<[u8]> for VRFPublicKey {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+impl Display for VRFPublicKey {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", hex::encode(self.0.to_bytes()))
+    }
+}
+
+impl VRFPublicKey {
+    pub fn vrf_verify(
```

Comment:
Hmm shouldn't we instead implement `VerifiableRandomFunction` trait?

Reply:
It had been updated already.